packages/vite/src/node/server/middlewares/htmlFallback.ts

```diff
@@ -16,10 +16,13 @@ export function htmlFallbackMiddleware(
       (req.method !== 'GET' && req.method !== 'HEAD') ||
       // Require Accept: text/html or */*
       !(
-        req.headers.accept === undefined || // equivalent to `Accept: */*`
-        req.headers.accept === '' || // equivalent to `Accept: */*`
-        req.headers.accept.includes('text/html') ||
-        req.headers.accept.includes('*/*')
+        (
+          req.headers.accept === undefined || // equivalent to `Accept: */*`
+          req.headers.accept === '' || // equivalent to `Accept: */*`
+          req.headers.accept.includes('text/html') ||
+          (req.headers.accept.includes('*/*') &&
+            !req.headers.accept.includes('image/*'))
+        ) // exclude favicon requests
```

Comment:
nit: the comment could be better place maybe (probably moved by prettier)

Reply:
Good catch! Thanks 🙏🏼